lib/execution/executor.ex

```diff
@@ -945,11 +1100,57 @@ defmodule WaspVM.Executor do
     |> Enum.count()
   end
 
-  defp rotl(number, shift) do
-    number <<< shift ||| number >>> (32 - shift)
+  defp rotl(number, shift), do: (number <<< shift) ||| (number >>> (0x1F &&& (32 + ~~~(shift + 1)))) &&& ~~~(0xFFFFFFFF <<< shift)
+  defp rotr(number, shift), do: (number >>> shift) ||| (number <<< (0x1F &&& (32 + ~~~(-shift + 1)))) &&& ~~~(0xFFFFFFFF <<< -shift)
+  def float_point_op(number) do
+    D.set_context(%D.Context{D.get_context | precision: 6})
```

Comment:
Why use decimals here instead of regular floats?

Reply:
The specs designated the floats had to be to a certain decimal. I worked off the Web Assembly Binary tool for a reference, regular floats I'm not 100% would satisfy it. Main thing is i wanted consistent output i.e not 0.0, and then 1.134567.